Topic1/key1/Cart.java

```diff
@@ -0,0 +1,38 @@
+package Topic1.key1;
+
+import java.util.ArrayList;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/**
+ * @author Jordan
+ * shopping cart
+ */
+public class Cart {
+
+	protected ArrayList<Item> products;
```

Comment:
Define products variable type as List instead of ArrayList.

Refer to "Program to an 'interface', not an 'implementation' principle.